agent/task.go

```diff
@@ -181,13 +181,24 @@ func (tc *taskContext) getCurrentCommand() command.Command {
 	return tc.currentCommand
 }
 
-func (tc *taskContext) setCurrentTimeout(dur time.Duration) {
+func (tc *taskContext) setCurrentTimeout(cmd command.Command) {
```

Comment:
it seems reasonable to refactor this so that you can test the "get timeout from command" from the task context structure, just so that the testing isn't so connected to state.

Reply:
I think this makes sense because it handles the nil timeout, task config idle timeout, and command cases. I think the right level of abstraction here is the task config.